ios/OpenTokReactNative/OTScreenCapture.m

```diff
@@ -30,117 +35,131 @@ - (instancetype)initWithView:(UIView *)view
+                                                metadata:nil];
+            // Signal for more frames
+            dispatch_semaphore_signal(self->_capturingSemaphore);
+        });
+    });
+}
+
+- (void)createCGContextFromPixelBuffer {
+    CGColorSpaceRef rgbColorSpace = CGColorSpaceCreateDeviceRGB();
+    CVPixelBufferLockBaseAddress(_pixelBuffer, 0);
     
-    NSParameterAssert(status == kCVReturnSuccess && _pixelBuffer != NULL);
+    _bitmapContext = NULL;
+    _bitmapContext = CGBitmapContextCreate(CVPixelBufferGetBaseAddress(_pixelBuffer),
+                                          CVPixelBufferGetWidth(_pixelBuffer),
+                                          CVPixelBufferGetHeight(_pixelBuffer),
+                                          8, CVPixelBufferGetBytesPerRow(_pixelBuffer), rgbColorSpace,
+                                          kCGBitmapByteOrder32Little | kCGImageAlphaPremultipliedFirst
+                                          );
+    CGContextTranslateCTM(_bitmapContext, 0.0f, self.view.frame.size.height);
+    CGContextScaleCTM(_bitmapContext, self.view.layer.contentsScale, -self.view.layer.contentsScale);
+    CVPixelBufferUnlockBaseAddress(_pixelBuffer, 0);
+    CFRelease(rgbColorSpace);
+}
 
+- (void)createPixelBuffer {
+
+    CFDictionaryRef ioSurfaceProps = CFDictionaryCreate( kCFAllocatorDefault, NULL, NULL, 0, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks );
+
+    NSDictionary *bufferAttributes = @{(id)kCVPixelBufferPixelFormatTypeKey : @(kCVPixelFormatType_32BGRA),
+                                       (id)kCVPixelBufferCGBitmapContextCompatibilityKey : @YES,
+                                       (id)kCVPixelBufferWidthKey : @(self.view.frame.size.width * _screenScale),
+                                       (id)kCVPixelBufferHeightKey : @(self.view.frame.size.height * _screenScale),
+                                       (id)kCVPixelBufferBytesPerRowAlignmentKey :
+                                           @(self.view.frame.size.width * _screenScale * 4),
+                                       (id)kCVPixelBufferIOSurfacePropertiesKey : (__bridge id)ioSurfaceProps
+                                       };
+     CVPixelBufferCreate(kCFAllocatorDefault,
+                                             self.view.frame.size.width,
+                                             self.view.frame.size.height,
+                                             kCVPixelFormatType_32ARGB,
+                                             (__bridge CFDictionaryRef)(bufferAttributes),
+                                          &_pixelBuffer);
+    CFRelease(ioSurfaceProps);
 }
 
 #pragma mark - Capture lifecycle
 
 /**
- * Allocate capture resources; in this case we're just setting up a timer and 
+ * Allocate capture resources; in this case we're just setting up a timer and
  * block to execute periodically to send video frames.
  */
+
 - (void)initCapture {
```

Comment:
being this empty maybe we should remove the funtion description

Reply:
Updated!